public/js/widget_config.js

```diff
@@ -0,0 +1,382 @@
+    )
+    .each((i, element) => {
+      // TODO: Provide a reasonable default value for element.value
+      if (element.value) {
+        const strPropSection = jQuery(element).data('section')
+        console.debug(strPropSection, element.name, element.value)
+        if (strPropSection === 'root') {
+          objNewWidget[element.name] = element.value
+        }
+        if (strPropSection === 'format') {
+          /*
+             * Some format values are integers.
+             */
+          const value = parseInt(element.value)
+          if (isNaN(value)) {
+            objNewWidget[strPropSection][element.name] = element.value
+          } else {
+            objNewWidget[strPropSection][element.name] = value
+          }
+        }
+        if (strPropSection === 'data') {
+          /*
+             * The topicAttribute element may contain multiple attributes.
+             * When found, assemble them into an Array of strings.
+             */
+          if (element.value.indexOf(RQ_PARAMS.ATTR_DELIMIT) > -1) {
+            const attributes = element.value
+              .replaceAll(' ', '')
+              .split(RQ_PARAMS.ATTR_DELIMIT)
+            objNewWidget[strPropSection][element.name] = attributes
+          } else {
+            objNewWidget[strPropSection][element.name] = element.value
+          }
+        }
+      }
+    })
+
+  // these are one off logic to string concat the values, not a nice 1-1 mapping
+  objNewWidget.position.my = `${jQuery('#widgetPositionMyX').val()} ${jQuery('#widgetPositionMyY').val()}`
+  objNewWidget.position.at = `${jQuery('#parentPositionAtX').val()} ${jQuery('#parentPositionAtY').val()}`
+
+  return objNewWidget
+}
+
+const initWidgetConfig = function (objSocket) { // eslint-disable-line no-unused-vars
+  socket = objSocket
+  /**
+     * Use the details of a new widget, collected via the configuration menu, to
+     * instantiate and position a new widget.
+     * It's called by clicking the "Create" button in the "Configure a New Widget"
+     * form which implies this function needs a better name.
+     */
+  // TODO: Give this function a more intuitive name based on its use and effect
+  const addWidget = function () {
+    const objNewWidget = extractWidgetConfigurationFromDialog()
+
+    objNewWidget.id = getNextId()
+
+    createWidget(objNewWidget)
+    positionWidgets()
+  }
+
+  jQuery('#newWidget').dialog({
+    width: 500,
+    autoOpen: false,
+    buttons: {
+      Create: addWidget,
+      Done: function () {
+        jQuery(this).dialog('close')
+      }
+    },
+    open: function (event, ui) {
+      jQuery('#menuDialog').dialog('close')
+    }
+  })
+
+  jQuery('#newWidget #newWidgetType').selectmenu({
+    change: function (event, ui) {
+      setNewWidgetDialogType(ui.item.value)
+    }
+  })
+
+  jQuery('#addWidget').on('click', function () {
+    jQuery('#newWidget').dialog({
+      title: 'Configure a New Widget',
```

Comment:
Since one activates this dialog by clicking the 'add widget' button, I believe a better, more intuitive, title would be 'Add a widget'. That will also be more consistent with the other title 'Reconfigure a widget'.